server/controllers/match/bedSearchController.ts

```diff
@@ -1,41 +1,37 @@
 import type { Request, RequestHandler, Response } from 'express'
+
 import { BedSearchParametersUi } from '../../@types/ui'
+import paths from '../../paths/match'
 import { PersonService } from '../../services'
 import BedService from '../../services/bedService'
 
+import { startDateFromParams } from '../../utils/matchUtils'
+
 export default class BedSearchController {
   constructor(private readonly bedService: BedService, private readonly personService: PersonService) {}
 
   search(): RequestHandler {
     return async (req: Request, res: Response) => {
       const query = Object.keys(req.query).length ? req.query : undefined
+      const body = Object.keys(req.body).length ? req.body : undefined
+
+      if (query || body) {
+        const params = {
+          ...query,
+          ...body,
+        }
+
+        params.startDate = startDateFromParams(params)
 
-      if (query) {
-        const {
-          durationDays,
-          maxDistanceMiles,
-          postcodeDistrict,
-          requiredPremisesCharacteristics,
-          requiredRoomCharacteristics,
-          startDate,
-          crn,
-        } = query
-
-        const bedSearchResults = await this.bedService.search(req.user.token, {
-          durationDays,
-          maxDistanceMiles,
-          postcodeDistrict,
-          requiredPremisesCharacteristics,
-          requiredRoomCharacteristics,
-          startDate,
-        } as BedSearchParametersUi)
-
-        const person = await this.personService.findByCrn(req.user.token, crn as string)
+        const bedSearchResults = await this.bedService.search(req.user.token, params as BedSearchParametersUi)
+        const person = await this.personService.findByCrn(req.user.token, params.crn as string)
 
         res.render('match/search', {
           pageHeading: 'Find a bed',
           bedSearchResults,
           person,
+          formPath: paths.beds.search({}),
```

Comment:
Is there a reason for passing this to the template instead of using it directly?

Reply:
Fair enough 👍